apps/core/src/hooks/useOnScreen.ts

```diff
@@ -1,32 +1,23 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { useState, useEffect, type MutableRefObject, useRef } from 'react';
+import { useState, useEffect, MutableRefObject } from 'react';
 
-export const useOnScreen = (ref: MutableRefObject<Element | null>) => {
+export const useOnScreen = (elementRef: MutableRefObject<Element | null>) => {
 	const [isIntersecting, setIsIntersecting] = useState(false);
 
-	const observerRef = useRef<IntersectionObserver>();
-	if (!observerRef.current) {
-		observerRef.current = new IntersectionObserver(
-			([entry]) => setIsIntersecting(entry.isIntersecting),
-			{
-				threshold: [0.01],
-			},
-		);
-	}
+	const update = ([entry]: IntersectionObserverEntry[]): void => {
+		setIsIntersecting(entry.isIntersecting);
+	};
 
 	useEffect(() => {
-		const currObserver = observerRef.current;
+		const node = elementRef.current;
+		if (!node) return;
 
-		if (ref.current && currObserver) {
-			currObserver.observe(ref.current);
-		}
-
-		return () => {
-			currObserver && currObserver.disconnect();
-		};
-	});
+		const observer = new IntersectionObserver(update, { threshold: 0.01 });
+		observer.observe(node);
+		return () => observer.disconnect();
+	}, [elementRef]);
```

Comment:
Does the linter not yell at you to specify `update` as a dependency of this `useEffect` since it's declared in the render code? The previous effect seems fine besides the mistake where we're calling it every single re-render

Reply:
good catch - not sure why the linter didn't pick up on this